data_safe_haven/commands/context.py

```diff
@@ -21,9 +24,9 @@ def show() -> None:
     """Show information about the selected context."""
     try:
         settings = ContextSettings.from_file()
-    except DataSafeHavenConfigError as exc:
-        print("No context configuration file.")
-        raise typer.Exit(code=1) from exc
+    except DataSafeHavenConfigError:
+        print("No context configuration file. Run `dsh context add` to create one.")
+        raise typer.Exit(code=1) from None
```

Comment:
Can you explain the reasoning for not inheriting from the `DataSafeHavenConfigError` exception?

Reply:
I suppose we could do, but I don't think it will have any effect here. The typer exception doesn't give a traceback or message, it just terminates the program with an exit code.

Reply:
I'd prefer either:

```python
except DataSafeHavenConfigError:
    ...
    raise typer.Exit(code=1)
```

or

```python
except DataSafeHavenConfigError as exc:
    ...
    raise typer.Exit(code=1) from exc
```

with no particular preference as to which is better

Reply:
Well, we can go with the second. The first is a linting error. The second feels a little pointless though because I don't think it has any effect.

Reply:
Why is raising from nothing a linting error but raising from `None` is fine? Seems a bit silly.

Reply:
:shrug: Explicit is better than implicit?

https://docs.astral.sh/ruff/rules/raise-without-from-inside-except/